### DISCUSSION TITLE
HTTP2 support

### DISCUSSION THREAD

**Discussion by siriuscor** [panva/openid-client#619] [Ideas]
Some OpenID servers only support HTTP/2. I'm wondering if there is any plan to support HTTP/2, or I can also submit a pull request.

**Comment by panva** [comment]
Hello, I am not opposed. How would such support look?

**Comment by siriuscor** [comment]
I added http2 for send request, and try to keep the original workflow, also I do some test on my own ,you can view the commit at
https://github.com/siriuscor/node-openid-client/commit/d39fc1766d9fec89f9687e102824d7a4f62522c4
user can add setting for http2 support
custom.setHttpOptionsDefaults({
  http2: true,
});

**Comment by NatoBoram** [comment]
I am experimenting with blocking HTTP/1.1 because that's what most bots are limited to.

A software I am using ([OpenID Connect auth plugin for PeerTube](https://framagit.org/framasoft/peertube/official-plugins/tree/master/peertube-plugin-auth-openid-connect)) is using this library, which is causing it to be blocked.

```json
{
	"level": "info",
	"ts": 1785536880.225794,
	"logger": "http.log.access.example.org",
	"msg": "handled request",
	"request": {
		"remote_ip": "104.20.26.136",
		"remote_port": "60330",
		"client_ip": "104.20.26.136",
		"proto": "HTTP/1.1",
		"method": "GET",
		"host": "authentik.example.org",
		"uri": "/application/o/peertube/.well-known/openid-configuration",
		"headers": {
			"Sec-Fetch-Mode": ["cors"],
			"Accept-Encoding": ["br, gzip, deflate"],
			"Connection": ["keep-alive"],
			"Accept": ["application/json"],
			"User-Agent": ["openid-client/v6.8.4"],
			"Accept-Language": ["*"]
		},
		"tls": {
			"resumed": false,
			"version": 772,
			"cipher_suite": 4865,
			"proto": "",
			"server_name": "authentik.example.org",
			"ech": false
		}
	},
	"bytes_read": 0,
	"user_id": "",
	"duration": 0.00004087,
	"size": 0,
	"status": 426,
	"resp_headers": {
		"Server": ["Caddy"],
		"Alt-Svc": ["h3=\":443\"; ma=2592000"],
		"Strict-Transport-Security": ["max-age=63072000"],
		"Connection": ["Upgrade"],
		"Upgrade": ["HTTP/3.0"]
	}
}
```
